modules/s3wrapper/src/s3wrapper/s3_client.py

```diff
@@ -0,0 +1,182 @@
+import re
+from datetime import timedelta
+
+from minio import Minio
+from minio.error import ResponseError
+
+
+class S3Client(object):
+    """ Wrapper around minio
+    """
+    
+    def __init__(self, endpoint, access_key=None, secret_key=None, secure=False):
+        self.__metadata_prefix = "x-amz-meta-"
+        self.client = None
+        try:
+            self.client = Minio(endpoint,
+                access_key=access_key,
+                secret_key=secret_key,
+                secure=secure)
+        except ResponseError as err:
+            print(err)
+
+    def __remove_objects_recursively(self, bucket_name):
+        objs = self.list_objects(bucket_name, recursive=True)
+        to_del = []
+        for obj in objs:
+            to_del.append(obj.object_name)
+
+        self.remove_objects(bucket_name, to_del)
+
+
+    def create_bucket(self, bucket_name, delete_contents_if_exists=False):
+        try:
+            if not self.exists_bucket(bucket_name):
+                self.client.make_bucket(bucket_name)
+            elif delete_contents_if_exists:
+                if self.__remove_objects_recursively(bucket_name):
+                    self.remove_bucket(bucket_name)
```

Comment:
should it not re/create the bucket after removing it? 
remove_bucket already calls __remove_objects_recursively but does not check the return value